### PR TITLE
Reconnect

### PR DESCRIPTION
**Pull request by richardscreene**

### CODE REVIEW

File: src/connection.js [hunk 1]
```diff
@@ -127,7 +136,32 @@ class Connection extends EventEmitter {
...
    */
-  _signalClose(graceful) {
+  async _signalClose(graceful) {
+
+    const reconnectTimeSeconds = this._config.getReconnectTimeSeconds();
+    if (!graceful && reconnectTimeSecs) {
+      try {
+        Logger.warn(`${LOG_NS} ${this.name} Connection has terminated - will attempt re-connection`);
+
+        await new Promise((resolve) => setTimeout(resolve, reconnectTimeSeconds * 1000));
```
Comment (connorsmallman) [inline src/connection.js:160]: how does it call itself again?

**Review by JonRobinson-Proximie — APPROVED** [review]

File: src/connection.js (hunk 1 above)
Comment (richardscreene) [reply src/connection.js:160]: the transport object calls this _signalClose() method on websocket error.

Comment (connorsmallman) [reply src/connection.js:160]: but does it call it every interval?

Comment (richardscreene) [reply src/connection.js:160]: I'm not sure what you mean by interval.
But, I did sneakily add a `isReconnecting` state.  Is that what you're getting at?

**Pull request merged**